### PR TITLE
Bug 1232694 - Make the secret volume for push/pull secrets unique

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@liggitt @csrwng PTAL

File: pkg/build/controller/strategy/util.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func setupSourceSecrets(pod *kapi.Pod, sourceSecret *kapi.LocalObjectReference)
 		return
 	}
 
-	mountSecretVolume(pod, sourceSecret.Name, sourceSecretMountPath)
+	mountSecretVolume(pod, sourceSecret.Name, sourceSecretMountPath, sourceSecret.Name)
```
Comment (liggitt) [inline pkg/build/controller/strategy/util.go]: `"source-"+sourceSecret.Name` for good measure?

**Comment by liggitt** [comment]
one name suggestion, LGTM otherwise

File: pkg/build/controller/strategy/util.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func mountSecretVolume(pod *kapi.Pod, secretName, mountPath string) {
 // allowing Docker to authenticate against private registries or Docker Hub.
 func setupDockerSecrets(pod *kapi.Pod, pushSecret, pullSecret *kapi.LocalObjectReference) {
 	if pushSecret != nil {
-		mountSecretVolume(pod, pushSecret.Name, DockerPushSecretMountPath)
+		mountSecretVolume(pod, pushSecret.Name, DockerPushSecretMountPath, "push-"+pushSecret.Name)
```
Comment (csrwng) [inline pkg/build/controller/strategy/util.go]: Can length of "push-" + pushSecret.Name be longer than a valid volume name ? 
If so, it'd be good to use util.namer.GetName() here

**Comment by csrwng** [comment]
just one comment, other than that looks good

**Comment by mfojtik** [comment]
@csrwng @liggitt i switched to namer.GetName(), thanks for suggesting this!

PTAL

**Comment by mfojtik** [comment]
[test]

**Comment by csrwng** [comment]
LGTM

**Comment by mfojtik** [comment]
[test]

**Comment by bparees** [comment]
[merge]

**Pull request merged**